src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/FilterRoleAssignmentsImpl.java

```diff
@@ -0,0 +1,139 @@
+    }
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseDataContent caseDataContent) {
+        log.info("Filter role assignments for case event {}", caseDataContent.getEvent().getEventId());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .collect(Collectors.toList());
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
+    }
+
+    @Override
+    public RoleAssignments filter(RoleAssignments roleAssignments, CaseTypeDefinition caseTypeDefinition) {
+        log.info("Filter role assignments for case type {}", caseTypeDefinition.getName());
+
+        List<RoleAssignment> roleAssignmentList = roleAssignments.getRoleAssignments().stream()
+            .filter(roleAssignment -> validateStartAndEndDate(roleAssignment))
+            .collect(Collectors.toList());
+        return RoleAssignments.builder().roleAssignments(roleAssignmentList).build();
+    }
+
+    private boolean validateJurisdiction(CaseDetails caseDetails, RoleAssignment roleAssignment) {
+        log.debug("Apply filter on jurisdiction {} for role assignment {}",
+            caseDetails.getJurisdiction(),
+            roleAssignment.getId());
+        roleAssignment.getMatchingResults().setValidJurisdiction(
+            roleAssignment.getAttributes().getJurisdiction()
+                .equals(caseDetails.getJurisdiction()));
+        return roleAssignment.getMatchingResults().isValidJurisdiction();
+    }
+
+    private boolean validateSecurityValidation(CaseDetails caseDetails, RoleAssignment roleAssignment) {
+        log.debug("Apply filter on security classification {} for role assignment {}",
+            roleAssignment.getClassification(),
+            roleAssignment.getId());
+        Optional<SecurityClassification> securityClassification = SecurityClassificationUtils
+            .getSecurityClassification(roleAssignment
+                .getClassification());
+        if (securityClassification.isPresent()) {
+            boolean value = caseHasClassificationEqualOrLowerThan(SecurityClassificationUtils
+                .getSecurityClassification(roleAssignment
+                    .getClassification())
+                .get())
+                .test(caseDetails);
+            roleAssignment.getMatchingResults().setValidClassification(value);
+        }
+        return roleAssignment.getMatchingResults().isValidClassification();
+    }
+
+    private boolean validateCaseId(CaseDetails caseDetails, RoleAssignment roleAssignment) {
+        log.debug("Apply filter on case id {} for role assignment {}",
+            roleAssignment.getAttributes().getCaseId(),
+            roleAssignment.getId());
+        roleAssignment.getMatchingResults().setValidCaseId(
+            roleAssignment.getAttributes().getCaseId().equals(caseDetails.getReferenceAsString()));
```

Comment:
Different types of role assignments can have different lists of attributes. Organisational role assignments for example won't have a `caseId` attribute
`roleAssignment.getAttributes().getCaseId().equals(caseDetails.getReferenceAsString()))` is going to throw a NullPointerException